feat: add fs_version and compressor output of nydus image check

PerseidMeteor opened the pull request:
## Relevant Issue (if applicable)
_If there are Issues related to this PullRequest, please list it._

## Details
1. Add fs_version value, output v5 or v6.
2. Add compressor algorithm value, like ztsd.
3. Add rafs_version and compressor JSON output of nydus image check, so that more info can be get if it is necessary.

## Types of changes

_What types of changes does your PullRequest introduce? Put an `x` in all the boxes that apply:_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Go over all the following points, and put an `x` in all the boxes that apply._

- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

imeoer commented on src/bin/nydus-image/main.rs:
```diff
@@ -114,6 +114,8 @@ impl OutputSerializer {
         blob_ids: Vec<String>,
         bootstrap: &Path,
+        compressor: String,
+        rafs_version: u32,
```
Should name it `fs_version`.

PerseidMeteor replied:
renamed

imeoer commented on src/bin/nydus-image/main.rs:
```diff
@@ -126,16 +128,14 @@ impl OutputSerializer {
+            out.insert("bootstrap".to_string(), serde_json::Value::from(bootstrap.display().to_string()));
+            out.insert("blobs".to_string(), serde_json::Value::from(blob_ids));
+            out.insert("trace".to_string(), serde_json::Value::from(root_tracer!().dump_summary_map().unwrap_or_default()));
+            out.insert("compressor".to_string(), serde_json::Value::from(compressor));
```
Let's we define the fields in `OutputSerializer`.

PerseidMeteor replied:
how about `fs_version` and `compressor`

imeoer replied:
> how about `fs_version` and `compressor`

Should be ok.

Desiki-high commented on src/bin/nydus-image/main.rs:
```diff
@@ -74,13 +77,19 @@ pub struct OutputSerializer {
     trace: serde_json::Map<String, serde_json::Value>,
+    /// Rafs version(v5 or v6).
+    fs_version: u32,
+    /// compress algorithm.
```
Capitalization: `Compress algorithm.`

PerseidMeteor replied:
resolved

imeoer commented on src/bin/nydus-image/main.rs:
```diff
@@ -74,13 +77,19 @@ pub struct OutputSerializer {
     blobs: Vec<String>,
     /// Performance trace info for current build.
     trace: serde_json::Map<String, serde_json::Value>,
+    /// Rafs version(v4, v5 or v6).
```
`Rafs version(v4, v5 or v6).` -> `Rafs version(5 or 6).`

imeoer replied:
In fact have no v4 for upstream codes. :)

PerseidMeteor replied:
fine

imeoer commented on src/bin/nydus-image/main.rs:
```diff
@@ -74,13 +77,19 @@ pub struct OutputSerializer {
     trace: serde_json::Map<String, serde_json::Value>,
+    /// Rafs version(v4, v5 or v6).
+    fs_version: u32,
+    /// Compress algorithm.
```
`Chunk compression algorithm`

imeoer commented on src/bin/nydus-image/main.rs:
```diff
@@ -95,11 +104,19 @@ impl OutputSerializer {
                 .with_context(|| format!("can not open output file {}", f.display()))?;
             let trace = root_tracer!().dump_summary_map().unwrap_or_default();
             let version = format!("{}-{}", build_info.package_ver, build_info.git_commit);
+            let out_fs_version: u32 = match fs_version {
```
Let's define `fs_version: RafsVersion` and use `RafsVersion::to_string()` here.

imeoer replied:
Let's define `compressor: compress::Algorithm` and use `compress::Algorithm::to_string()` here.

imeoer replied:
Or implement JSON Serialize for `RafsVersion` and `compress::Algorithm`.

PerseidMeteor replied:
we can use `to_string` method

imeoer commented on src/bin/nydus-image/main.rs:
```diff
@@ -74,13 +74,19 @@ pub struct OutputSerializer {
     blobs: Vec<String>,
     /// Performance trace info for current build.
     trace: serde_json::Map<String, serde_json::Value>,
+    /// Rafs version(v5 or v6).
```
Should be `5 or 6`

PerseidMeteor replied:
resolved

imeoer reviewed (commented):
other LGTM, thanks!

PerseidMeteor committed:
feat: add fs_version and compressor output of nydus image check
1.Add rafs_version value, output like 5 or 6.
2.Add compressor algorithm value, like ztsd.
Add rafs_version and compressor json output of nydus image check,so that more info can be get if it is necessary.

Signed-off-by: YuQiang <y_q_email@163.com>

adam3q reviewed (commented):
Thanks! Could you update the `nydus-image check` usage in `docs/nydus-image.md` ?

imeoer reviewed (approved):
Thanks!

PerseidMeteor commented:
> Thanks! Could you update the `nydus-image check` usage in `docs/nydus-image.md` ?

`docs/nydus-image.md`  will be updated later.
